modin/pandas/base.py

```diff
@@ -3665,6 +3666,95 @@ def __getitem__(self, key):
         else:
             return self._getitem(key)
 
+    def xs(
+        self,
+        key,
+        axis=0,
+        level=None,
+        drop_level: bool = True,
+    ):  # noqa: PR01, RT01, D200
+        """
+        Return cross-section from the Series/DataFrame.
+        """
+        axis = self._get_axis_number(axis)
+        labels = self.axes[axis]
+
+        if isinstance(key, list):
+            # deprecated in pandas, to be removed in 2.0
+            warnings.warn(
+                "Passing lists as key for xs is deprecated and will be removed in a "
+                + "future version. Pass key as a tuple instead.",
+                FutureWarning,
+            )
+
+        if level is not None:
+            if not isinstance(labels, pandas.MultiIndex):
+                raise TypeError("Index must be a MultiIndex")
+            loc, new_ax = labels.get_loc_level(key, level=level, drop_level=drop_level)
+
+            # create the tuple of the indexer
+            _indexer = [slice(None)] * self.ndim
+            _indexer[axis] = loc
+            indexer = tuple(_indexer)
+
+            result = self.iloc[indexer]
+            setattr(result, self._pandas_class._get_axis_name(axis), new_ax)
+            return result
+
+        if axis == 1:
+            if drop_level:
+                return self[key]
+            index = self.columns
+        else:
+            index = self.index
+
+        new_index = None
+        if isinstance(index, pandas.MultiIndex):
+            loc, new_index = index._get_loc_level(key, level=0)
+            if not drop_level:
+                if is_integer(loc):
+                    new_index = index[loc : loc + 1]
+                else:
+                    new_index = index[loc]
```

Comment:
```suggestion
            loc, new_index = index.get_loc_level(key, level=0, drop_level=drop_level)
```
comparing this piece of code with a non-private [`get_loc_level`](https://github.com/pandas-dev/pandas/blob/2e218d10984e9919f0296931d92ea851c6a6faf5/pandas/core/indexes/multi.py#L3014-L3025) method from pandas it seems that we're just copying its implementation here, why is that? why can't we use the public version of this method? @jbrockmendel

Reply:
get_loc_level can take either a level label or number, which can lead to ambiguity. _get_loc_level is specifically a number.

Reply:
@dchigarev any other thoughts on that?